NDTensors/src/lib/UnallocatedArrays/src/abstractfill/set_types.jl

```diff
@@ -0,0 +1,77 @@
+using FillArrays: AbstractFill, Fill, Zeros
+using NDTensors.SetParameters: SetParameters, Position
+using NDTensors.UnspecifiedTypes: UnspecifiedZero
+
+# `SetParameters.jl` overloads.
+SetParameters.get_parameter(::Type{<:AbstractFill{P1}}, ::Position{1}) where {P1} = P1
+SetParameters.get_parameter(::Type{<:AbstractFill{<:Any,P2}}, ::Position{2}) where {P2} = P2
+function SetParameters.get_parameter(
+  ::Type{<:AbstractFill{<:Any,<:Any,P3}}, ::Position{3}
+) where {P3}
+  return P3
+end
+
+## Setting paramaters
+# Set parameter 1
+function SetParameters.set_parameter(T::Type{<:AbstractFill}, ::Position{1}, P1)
```

Comment:
So defining these functions on the type `AbstractFill` is not type piracy because the function is defined in `SetParameters` and we manage that packages. In contrast to, say, overloading a `Base` function on the `AbstractFill` type where we manage neither `Base` nor `AbstractFill`

Reply:
That's basically right, though `UnallocatedArrays` is probably not the right place to define this. If `SetParameters` was a registered package, these definitions would make the most sense to go into a `SetParametersFillArraysExt` package extension, since it is extending functionality of `SetParameters` for types defined in `FillArrays`. Probably we should move them to `src/lib/SetParameters/ext/SetParametersFillArraysExt` to anticipate how we would organize the code if these were all registered packages.

Reply:
This makes sense! I moved this to a `SetParametersFillArraysExt` folder, made a `Project.toml` and started to set up `SetParameters` for extensions.

Reply:
Sounds good, thanks.